### PR TITLE
SWATCH-2444: Fix bug with multiple azure API credentials 

### PR DESCRIPTION
**Pull request by kflahert**
Jira issue: SWATCH-2444

## Description

When we have multiple sets of api credentials to use against azure, the code was calling the api with both credentials every time. This code change makes it so we only try the next set of credentials if the first one fails instead of continuing to try more credentials.

To test this you need to use credentials from stage, updated to have another invalid entry.

### CODE REVIEW

File: swatch-producer-azure/src/test/java/com/redhat/swatch/azure/service/AzureMarketplaceServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import com.redhat.swatch.clients.azure.marketplace.api.resources.AzureMarketplaceApi;
+import io.quarkus.test.InjectMock;
+import io.quarkus.test.junit.QuarkusTest;
+import jakarta.ws.rs.ProcessingException;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+@QuarkusTest
```
Comment (Sgitario) [inline swatch-producer-azure/src/test/java/com/redhat/swatch/azure/service/AzureMarketplaceServiceTest.java:44]: I don't think this test needs to use `@QuarkusTest`.

File: swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceClientFactory.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    this.azureMarketplaceCredentials = azureMarketplaceCredentials;
+    this.oidcClients = oidcClients;
+  }
+
+  public List<AzureMarketplaceApi> createClientForEachTenant() {
+    return azureMarketplaceCredentials.getClients().stream()
+        .map(
+            client -> {
+              try {
+                return RestClientBuilder.newBuilder()
```
Comment (Sgitario) [inline swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceClientFactory.java:61]: Use QuarkusRestClientBuilder instead (this is the recommended builder for Quarkus Reactive REST Client).

**Review by Sgitario — CHANGES_REQUESTED** [review]
Very well spotted! 
Only a few minor things in my review.

**Comment by kflahert** [comment]
> Very well spotted! Only a few minor things in my review.

Thanks

**Review by Sgitario — APPROVED** [review]

**Pull request merged**